pkg/e2e/verify/namespace_webhook.go

```diff
@@ -124,6 +128,13 @@ var _ = ginkgo.Describe(namespaceWebhookTestName, func() {
 				return true, nil
 			})
 		})
+		
+		ginkgo.JustAfterEach(func() {
+			h.Impersonate(rest.ImpersonationConfig{})
+			err := deletePrometheusRule(namespace, prometheusName, h)
```

Comment:
So the goal of this function will be to clean up any PrometheusRules the tests might have created. In the tests currently, they are attempting to create PrometheusRules in a whole bunch of different namespaces (anything in `PRIVILEGED_NAMESPACES` or `NONPRIV_NAMESPACES`

Which means that the cleanup function should attempt to clean up any PromRule named `prometheusName` in every namespace in those lists.

Now naturally it is not going to exist in some of them, and that's fine - nothing to clean up! But that will be raised as an error by the delete.

So rather than treating every non-nil err returned as a test failure, we should be checking if the error is a "good" error. In this case, something like:

```
if err != nil and !errors.IsNotFound(err) {
   // this is not good and means that it was an error and wasn't due to the object not existing
}
```

(in this case, the `errors` package is `k8s.io/apimachinery/pkg/api/errors` which would need to be added to the imports.

Reply:
Another thing: because this `JustAfterEach` is only relevant to the tests doing PromRule stuff, maybe all the PromRule tests can be packaged inside a nested `ginkgo.Describe("xxx", func(){  }` block .. along with this.. that way it should only execute for the promrule tests.